cfavml/src/danger/impl_avx2.rs

```diff
@@ -2724,3 +2777,55 @@ impl SimdRegister<u64> for Avx2 {
         _mm256_storeu_si256(mem.cast(), reg)
     }
 }
+
+impl Hypot<f64> for Avx2 {
+    #[inline(always)]
+    unsafe fn hypot(x: Self::Register, y: Self::Register) -> Self::Register {
+        // convert the inputs to absolute values
+        let (x, y) = (
+            _mm256_andnot_pd(_mm256_set1_pd(-0.0), x),
+            _mm256_andnot_pd(_mm256_set1_pd(-0.0), y),
+        );
+        // find the max and min of the two inputs
+        let (hi, lo) = (
+            <Avx2 as SimdRegister<f64>>::max(x, y),
+            <Avx2 as SimdRegister<f64>>::min(x, y),
+        );
```

Comment:
Personally, I try and avoid stacking ops like this into a tuple, it is easier to read IMO if they're just 2 lines rather than 4 and reading from top down rather than jumping about between declaration and op itself.